editor/plugins/animation_library_editor.h

```diff
@@ -0,0 +1,120 @@
+/* Copyright (c) 2007-2022 Juan Linietsky, Ariel Manzur.                 */
+/* Copyright (c) 2014-2022 Godot Engine contributors (cf. AUTHORS.md).   */
+/*                                                                       */
+/* Permission is hereby granted, free of charge, to any person obtaining */
+/* a copy of this software and associated documentation files (the       */
+/* "Software"), to deal in the Software without restriction, including   */
+/* without limitation the rights to use, copy, modify, merge, publish,   */
+/* distribute, sublicense, and/or sell copies of the Software, and to    */
+/* permit persons to whom the Software is furnished to do so, subject to */
+/* the following conditions:                                             */
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#ifndef ANIMATION_LIBRARY_EDITOR_H
+#define ANIMATION_LIBRARY_EDITOR_H
+
+#include "editor/animation_track_editor.h"
+#include "editor/editor_plugin.h"
+#include "scene/animation/animation_player.h"
+#include "scene/gui/dialogs.h"
+#include "scene/gui/tree.h"
+
+class EditorFileDialog;
+
+class AnimationLibraryEditor : public AcceptDialog {
+	GDCLASS(AnimationLibraryEditor, AcceptDialog)
+
+	enum {
+		LIB_BUTTON_ADD,
+		LIB_BUTTON_LOAD,
+		LIB_BUTTON_PASTE,
+		LIB_BUTTON_FILE,
+		LIB_BUTTON_DELETE,
+	};
+	enum {
+		ANIM_BUTTON_COPY,
+		ANIM_BUTTON_FILE,
+		ANIM_BUTTON_DELETE,
+
+	};
+
+	enum FileMenuAction {
+		FILE_MENU_SAVE_LIBRARY,
+		FILE_MENU_SAVE_AS_LIBRARY,
+		FILE_MENU_MAKE_LIBRARY_UNIQUE,
+		FILE_MENU_EDIT_LIBRARY,
+
+		FILE_MENU_SAVE_ANIMATION,
+		FILE_MENU_SAVE_AS_ANIMATION,
+		FILE_MENU_MAKE_ANIMATION_UNIQUE,
+		FILE_MENU_EDIT_ANIMATION,
+	};
+
+	enum FileDialogAction {
+		FILE_DIALOG_ACTION_OPEN_LIBRARY,
+		FILE_DIALOG_ACTION_SAVE_LIBRARY,
+		FILE_DIALOG_ACTION_OPEN_ANIMATION,
+		FILE_DIALOG_ACTION_SAVE_ANIMATION,
+	};
+
+	FileDialogAction file_dialog_action = FILE_DIALOG_ACTION_OPEN_ANIMATION;
+
+	StringName file_dialog_animation;
+	StringName file_dialog_library;
+
+	AcceptDialog *error_dialog;
```

Comment:
A few members are missing a `nullptr` init here.

Reply:
oh, I realize I started this before your work on adding nullptr, will keep this present next time.